src/blocks/placeholders/index.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { registerBlockType } from '@wordpress/blocks';
+
+/**
+ * Placeholder blocks.
+ */
+const placeholders = [
+	{
+		name: 'site-title',
+		title: __( 'Site Title', 'woocommerce-gutenberg-emails' ),
+		description: __( 'Display the site title.', 'woocommerce-gutenberg-emails' ),
+	},
+	{
+		name: 'order-date',
+		title: __( 'Order Date', 'woocommerce-gutenberg-emails' ),
+		description: __( 'Display the date the order was placed.', 'woocommerce-gutenberg-emails' ),
+	},
+	{
+		name: 'order-number',
+		title: __( 'Order Number', 'woocommerce-gutenberg-emails' ),
+		description: __( 'Display the order number.', 'woocommerce-gutenberg-emails' ),
+	},
+	{
+		name: 'order-billing-full-name',
+		title: __( 'Billing Full Name', 'woocommerce-gutenberg-emails' ),
+		description: __( 'Display the full billing name from the order.', 'woocommerce-gutenberg-emails' ),
+	},
+	{
+		name: 'user-pass',
+		title: __( 'User Password', 'woocommerce-gutenberg-emails' ),
+		description: __( 'Display the user password.', 'woocommerce-gutenberg-emails' ),
+	},
+	{
+		name: 'user-login',
+		title: __( 'User Login', 'woocommerce-gutenberg-emails' ),
+		description: __( 'Display the user login name.', 'woocommerce-gutenberg-emails' ),
+	},
+	{
+		name: 'lost-password-url',
+		title: __( 'Lost Password URL', 'woocommerce-gutenberg-emails' ),
+		description: __( 'Display the URL to the lost password page.', 'woocommerce-gutenberg-emails' ),
+	},
+	{
+		name: 'user-email',
+		title: __( 'User Email', 'woocommerce-gutenberg-emails' ),
+		description: __( 'Display the user email address.', 'woocommerce-gutenberg-emails' ),
+	},
+	{
+		name: 'password-generateod',
+		title: __( 'Generated Password', 'woocommerce-gutenberg-emails' ),
+		description: __( 'Display the randomly generated password for the user.', 'woocommerce-gutenberg-emails' ),
+	},
+	{
+		name: 'customer-note',
+		title: __( 'Customer Note', 'woocommerce-gutenberg-emails' ),
+		description: __( 'Display the added customer note.', 'woocommerce-gutenberg-emails' ),
+	},
+];
+
+/**
+ * Register each of the placeholder blocks.
+ */
+placeholders.forEach( ( placeholder ) => {
+	registerBlockType( 'woocommerce-gutenberg-emails/' + placeholder.name, {
+		title: placeholder.title,
+		category: 'woocommerce-gutenberg-emails',
+		keywords: [ __( 'WooCommerce', 'woocommerce-gutenberg-emails' ) ],
+		description: placeholder.description,
+
+		/**
+		 * Renders and manages the block.
+		 *
+		 * @return {Object} Editor component.
+		 */
+		edit() {
+			return (
+				<span className="placeholder">
+					{ '{' + placeholder.name.replace( '-', '_' ) + '}' }
+				</span>
+			);
+		},
+
+		/**
+		 * Block content is rendered in PHP, not via save function.
```

Comment:
Oh, just noticed this comment is wrong, the content is not rendered in PHP. It should be changed in [`order-details` block](https://github.com/Aljullu/woocommerce-gutenberg-emails/blob/master/src/blocks/order-details/index.js#L38) as well. I think we can just copy the comment from the WooCommerce product blocks for now:
```suggestion
		 * Save the block content in the post content.
```